src/sqlfluff/core/parser/segments/base.py

```diff
@@ -1238,7 +1238,7 @@ def _validate_segment_after_fixes(self, rule_code, dialect, fixes_applied, segme
 
     @staticmethod
     def _log_apply_fixes_check_issue(message, *args):  # pragma: no cover
-        linter_logger.critical(message, *args)
+        linter_logger.critical(message, exc_info=True, *args)
```

Comment:
This provides a call stack showing exactly where the parse error occurred. Very helpful for diagnosing issues like this.